File: packages/transform/src/features/protobuf/ProtobufAssertEncodeTransformer.ts
import { ProtobufAssertEncodeProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufAssertEncodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "protobuf.assertEncode",
      write: ProtobufAssertEncodeProgrammer.write,
    });
}

File: packages/transform/src/features/protobuf/ProtobufAssertEncodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufAssertEncodeTransformer = protobufAssertEncodeTransformerNamespace{}

type protobufAssertEncodeTransformerNamespace struct{}

func (protobufAssertEncodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufAssertEncodeProgrammer.Write(call.TypeText)
}
